### PR TITLE
:sparkles: (Dockfile) Upgrade Dockerfile

### PR DESCRIPTION
**Pull request by xiekeyi98**
1. bump golang from 1.17 to 1.18. 2. use buidkit for docker build.

breaking: build docker with buildkit for performance,but it require
18.09 or
   higher(https://docs.docker.com/develop/develop-images/build_enhancements/)

### CODE REVIEW

**Comment by xiekeyi98** [comment]
image

File: resources/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,16 @@
-FROM golang:1.17-alpine as builder
+FROM golang:1.18-alpine as builder
```
Comment (haoel) [inline resources/Dockerfile:1]: I remember I updated this yesterday in #18, anyway, not sure why have this diff..

Comment (xiekeyi98) [reply resources/Dockerfile:1]: sorry, I think cuz I forgot to trace the upstream.

**Review by haoel — APPROVED** [review]
thanks for this contribution!

**Pull request merged**